### PR TITLE
Gfo 13

### PR DESCRIPTION
**Pull request by LLaszlo89**
Well let's get started.

### CODE REVIEW

File: backend/migrations/20200808174929-Orders.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    ordered_at:{type:'timestamp'},
+  });
+};
+exports.down = function(db) {
+  return db.dropTable('Orders');
+};
+
+exports._meta = {
+  "version": 1
+};
```
Comment (marcianszlovak) [inline backend/migrations/20200808174929-Orders.js:62]: Rename file to Order so naming is singular like Ticket?

File: backend/test/authenticate.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const mockResponse = () => {
+      const res = {};
+      res.status = 401;
+      res.json = 'hi';
+      return res;
+    };
+
+    const resp = mockResponse();
+    authenticateMiddleware.authenticate(invalid_input, resp, next);
+    expect(resp.status).toBe(401);
```
Comment (adamgyulavari) [inline backend/test/authenticate.test.js:22]: @LLaszlo89 
- here you don't need to mock the `next` that won't be used, you can just pass a `null`
- `resp` on the other hand need to contain a `status()` that will be _called_ and that should be checked
- also the thing that `status()` is returning should have a `json()` that will be _called_ and that also should be checked

File: backend/test/session.test.js [hunk 3]
```diff
@@ -1,21 +1,63 @@
...
+    );
+    jwt.sign.mockImplementation((data, sec) => {
+      return data;
+    });
+
+    await request(app)
       .post('/api/session')
       .set('Content-Type', 'application/json')
-      .send({ name: 'Lehel', password: 'asdf' })
+      .send(InvalideUser)
```
Comment (adamgyulavari) [inline backend/test/session.test.js:31]: @LLaszlo89 either fix the spelling and variable names start with lowerCase letters or just put here the object

File: backend/test/session.test.js [hunk 4]
```diff
@@ -1,21 +1,63 @@
...
+    };
+    const response = { results: [userResult] };
 
+    UserRepository.prototype.getUser.mockImplementation(() =>
+      Promise.resolve(response)
+    );
+
+   const x = await request(app)
+      .post('/api/session')
+      .send(ValideUser)
```
Comment (adamgyulavari) [inline backend/test/session.test.js:53]: @LLaszlo89 same here

File: backend/test/subscription.test.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { UserService } from '../src/services/UserService';
 
 
 jest.mock('../src/repository/SubscriberRepository');
-jest.mock('../src/services/UserService');
+jest.mock('../src/services/UserService');
```
Comment (adamgyulavari) [inline backend/test/subscription.test.js:9]: @LLaszlo89 remove the space at the end

File: backend/test/subscription.test.js [hunk 6]
```diff
@@ -42,3 +42,4 @@ describe('testing /api/subscription endpoint', () => {
     expect(mockRegister.body).toEqual(validSubscriber);
   });
 });
+
```
Comment (adamgyulavari) [inline backend/test/subscription.test.js:45]: @LLaszlo89 empty line

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

File: backend/src/repository/TicketTypesRepository.js [hunk 7]
```diff
@@ -1,26 +1,29 @@
...
-  constructor() {}
-  async saveTicket(ticket) {
+export class TicketTypeRepository {
+  constructor() {
+    this.db = process.env.MYSQL_DATABASE;
+  }
+  async saveTicketType(ticket) {
     await db.query(
-      `INSERT INTO foxticket.TicketTypes (name, price, description, icon)
+      `INSERT INTO ${this.db}.TicketTypes (name, price, description, icon)
```
Comment (adamgyulavari) [inline backend/src/repository/TicketTypesRepository.js:10]: @LLaszlo89 just leave out the db:
```suggestion
      `INSERT INTO TicketTypes (name, price, description, icon)
```

File: backend/src/seed.js [hunk 8]
```diff
@@ -1,26 +1,30 @@
...
-let userDummyData = {
+userRepo.save({
   name: 'Lehel',
   email: 'lehel@gmail.com',
-  password: 'password123',
-};
-let adminDummyData = {
+  password: '$2b$10$H7mQGKBAO66DkB2CY49LzujvzHeO9mB3OheHwzDvKfG7RlgTOq4Vm',
+});
+userRepo.saveAdmin({
```
Comment (adamgyulavari) [inline backend/src/seed.js:11]: @LLaszlo89 ugh, I've missed this :)
we should have only one method for saving the user, the difference is the user object (that can be admin or not admin shouldn't matter)

**Review by adamgyulavari — CHANGES_REQUESTED** [review]

**Review by adamgyulavari — APPROVED** [review]

**Pull request merged**